jnigen/lib/src/bindings/dart_generator.dart

```diff
@@ -1131,3 +1231,141 @@ class _JValueWrapper extends TypeVisitor<String> {
     return '$_jni.JValue${node.name.capitalize()}($param)';
   }
 }
+
+/// Builds access expressions from the end to the beginning.
+///
+/// Acts as [StringBuffer] that prepends instead of appending on write.
+/// It also automatically adds the appropriate number of opening parenthesis
+/// at the beginning when calling [toString].
+///
+/// For example:
+/// ```dart
+/// final exprBuilder = ReverseExpressionBuilder();
+/// exprBuilder.write(' as A).first');
+/// exprBuilder.write(' as B).second');
+/// exprBuilder.write('third');
+/// exprBuilder.toString(); // ((third as B).second as A).first
+/// ```
+@visibleForTesting
+class ReverseExpressionBuilder {
```

Comment:
This still feels like a weird abstraction to me. I would almost prefer a Dart AST instead, but that would make it horribly verbose.

I'm happy to just land this.